nixos/modules/services/misc/ethminer.nix

```diff
@@ -0,0 +1,116 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ethminer;
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.ethminer = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "enable ethminer ether mining";
+      };
+
+      recheck = mkOption {
+        type = types.string;
+        default = "1000";
+        description = "time to recheck";
+      };
+
+      toolkit = mkOption {
+        type = types.string;
+        default = "cuda";
+        description = "cuda or opencl";
+      };
+
+      apiPort = mkOption {
+        type = types.string;
+        default = "-3333";
```

Comment:
A port starting with a minus ?

Reply:
minus indicated read-only mode for api.  indicated in description.